api/src/OpenTelemetry/Internal/Common/Types.hs

```diff
@@ -83,6 +108,25 @@ data FlushResult
   deriving (Show)
 
 
+-- | Returns @FlushError@ if either argument is @FlushError@, @FlushTimeout@ if either argument is @FlushTimeout@, and @FlushSuccess@ otherwise.
+takeWorseFlushResult :: FlushResult -> FlushResult -> FlushResult
+takeWorseFlushResult FlushError _ = FlushError
+takeWorseFlushResult _ FlushError = FlushError
+takeWorseFlushResult FlushTimeout _ = FlushTimeout
+takeWorseFlushResult _ FlushTimeout = FlushTimeout
+takeWorseFlushResult _ _ = FlushSuccess
+
+
+takeWorstFlushResult :: (Foldable t) => t FlushResult -> FlushResult
+takeWorstFlushResult = foldr takeWorseFlushResult FlushSuccess
+
+
+exportResultToFlushResult :: ExportResult -> FlushResult
+exportResultToFlushResult Success = FlushSuccess
+exportResultToFlushResult (Failure _) = FlushError
```

Comment:
I don't know of a better way to do this conversion than drop the Exception

Reply:
```suggestion
exportResultToFlushResult (Failure mErr) = FlushError $ toList mErr
```